osdep/win32-console-wrapper.c

```diff
@@ -38,26 +38,31 @@ static void cr_perror(const wchar_t *prefix)
 
 static int cr_runproc(wchar_t *name, wchar_t *cmdline)
 {
-    STARTUPINFOW si;
-    STARTUPINFOW our_si;
-    PROCESS_INFORMATION pi;
     DWORD retval = 1;
 
-    ZeroMemory(&si, sizeof(si));
-    si.cb = sizeof(si);
-    si.hStdInput = GetStdHandle(STD_INPUT_HANDLE);
-    si.hStdOutput = GetStdHandle(STD_OUTPUT_HANDLE);
-    si.hStdError = GetStdHandle(STD_ERROR_HANDLE);
-    si.dwFlags |= STARTF_USESTDHANDLES;
-
     // Copy the list of inherited CRT file descriptors to the new process
-    our_si.cb = sizeof(our_si);
+    STARTUPINFOW our_si = {sizeof(our_si)};
     GetStartupInfoW(&our_si);
-    si.lpReserved2 = our_si.lpReserved2;
-    si.cbReserved2 = our_si.cbReserved2;
-
-    ZeroMemory(&pi, sizeof(pi));
 
+    // Don't redirect std streams if they are attached to a console. Let mpv
+    // attach to the console directly in this case. In theory, it should work
+    // out of the box because "console-like" handles should be managed by Windows
+    // internally, which works for INPUT and OUTPUT, but in certain cases,
+    // not for ERROR.
+    DWORD mode;
+    HANDLE hStdInput  = GetStdHandle(STD_INPUT_HANDLE);
+    HANDLE hStdOutput = GetStdHandle(STD_OUTPUT_HANDLE);
+    HANDLE hStdError  = GetStdHandle(STD_ERROR_HANDLE);
+    STARTUPINFOW si = {
+        .cb          = sizeof(si),
+        .lpReserved2 = our_si.lpReserved2,
+        .cbReserved2 = our_si.cbReserved2,
```

Comment:
I know this was already here, but I don't think there is a point of copying these values (and thus, no need for `our_si` structure and `GetStartupInfoW` call): If you observe these values, they are always zero/null. But even if they weren't, what's the point of copying them? The documentation also says that they must be zero/null when setting the STARTUPINFO structure.

Reply:
See the commit that added this https://github.com/mpv-player/mpv/commit/c19f634e6cbdeaffed1d56ecd3a31c0652820cdf it is explained why it is used. Sure those are reserved fields, but apparently they are used for those things.

Reply:
We could pass through whole `GetStartupInfoW`, but I don't see a use case for that. we just want to start mpv.exe with console. If anyone needs to pass more params, they should use mpv.exe directly.